website: category reorganization

oliverpool opened the pull request:
fixes #126

This is what I currently have:
```
1. Quickstart (unchanged)
2. Supported platforms (Hardware and CI)
3. Configuration
  - package config (unchanged)
  - remote syslog (unchanged)
  - TLS (unchanged)
  - wifi (unchanged)
  - bluetooth (unchanged)
  - Access permanent data (unchanged)
4. Available packages
  - Showcase (unchanged)
  - tailscale (unchanged)
  - caddy (unchanged)
  - min.io (unchanged)
5. Package development
  - process interface (unchanged)
  - input/output (unchanged)
  - go modules (unchanged)
  - Non-Go Prototyping (unchanged)
```

Compared to my initial suggestion, here are the adjustments:
- "Available packages" is a new page, with a short explanation on how to add go commands
- removal of the category "Packaging new software" (content is in "Available packages")
- “Showcase” a sub page under “Available packages” (as you suggested)
- "Non-Go Prototyping" is a sub-page of "Package development" (name as you suggested, a bit more hidden as a sub-page, since it’s "supposed to be a stop-gap")
- bluetooth is under "configuration" (as you suggested)

I did some content changes, but tried to limit them as much as possible, to tackle first the category structure.

Once you are happy with it, we can see if some text need to be updated.

oliverpool commented:
I just realized, that my code editor (vscode) did some auto-formatting on the markdown content... Should I revert them?

stapelberg commented on website/content/packages/_index.markdown:
```diff
@@ -0,0 +1,29 @@
+If you feel a common program should be included here but isn’t, please [file an
+issue on GitHub](https://github.com/gokrazy/gokrazy/issues).
+
+## Installing existing Go command
```
Go command → Go software

stapelberg commented on website/content/prototyping/_index.markdown:
```diff
@@ -1,195 +1,17 @@
 
-This article shows a couple of techniques for getting non-Go
-software to work on gokrazy, in increasing order of complexity.
+A gokrazy package is simply the `main` package of a Go software.
```
That’s not entirely correct, actually. A gokrazy package is a Go package.

Even if the Go package declares a named library instead of a `main` package, that’s still a valid gokrazy package, and can be useful to install. https://github.com/gokrazy/podman is an example for that: it declares `package podman` in `podman.go`, so it won’t result in an extra binary (no `/user/podman`). But, because it also contains a `_gokrazy/extrafiles.tar`, those files will be included in the root file system.

stapelberg commented on website/content/prototyping/_index.markdown:
```diff
@@ -1,195 +1,17 @@
-The Prometheus node-exporter doesn’t use cgo and needs no
-command-line parameters, configuration files or other assets can be
-added to the <code>gokr-packer</code> command line.
+To use the local version of a repository, simply use the [replace directive](https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive) of `go.mod`.
```
Here and above, remove “simply” — it’s subjective and doesn’t add anything :)

stapelberg commented on website/content/userguide/_index.markdown:
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Userguide"
+title: "Configuration"
```
The rest of this file still mentions “Userguide”. That should probably be updated to “Configuration” as well.

stapelberg reviewed (changes requested):
Thanks for this PR!

Can you add `aliases:` directives to all files that you moved? That way, the links should keep working.

I don’t mind the markdown formatting changes, so no need to revert them.

oliverpool commented:
Thanks for the review!

I have added `aliases` to all the page that were (re)moved, I think I didn't miss any.

I also replaced `userguide` with `configuration guides`.

stapelberg commented on website/content/prototyping/_index.markdown:
```diff
@@ -1,195 +1,17 @@
 
-This article shows a couple of techniques for getting non-Go
-software to work on gokrazy, in increasing order of complexity.
+A gokrazy package is simply a package of a Go software.
```
We can make this a bit more tangible. How about:

A gokrazy package corresponds to a Go module. The `gokr-packer` program is a wrapper around `go install`.

oliverpool replied:
> A gokrazy package corresponds to a Go module. 

I think a Go module can have multiple packages, so this is also not optimal.

For instance the `github.com/VictoriaMetrics/VictoriaMetrics` module as many packages: https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/app

stapelberg replied:
Right, let’s make it:

We can make this a bit more tangible. How about:

A gokrazy package corresponds to a Go package (which in turn is part of a Go module). The `gokr-packer` program is a wrapper around `go install`.

stapelberg reviewed (changes requested):
Thanks, almost there :)

oliverpool commented on website/content/prototyping/_index.markdown:
```diff
@@ -1,195 +1,17 @@
 ---
 layout: default
-title: Non-Go Prototyping
+title: Package development
```
I kept the `/prototyping/` URI, since the current `Non-Go Prototyping` page actually starts with go sections:
![2022-04-10-134058](https://user-images.githubusercontent.com/3864879/162616348-4ce63857-db44-4c43-940d-9fce4c21f6a9.png)

The `Go software not written for gokrazy:` remains available at this URI (only the non-go stuff changed location).

I am not found of using `/development/` for this. I think it would be better to document how to contribute to gokrazy under this URI (supervisor, packer...).

What do you think?

stapelberg replied:
I think `/contributing/` would be better for documentation about contributing :) In general, the URL should match the title.

I understand the split between Go and non-Go software, but I still think the `/prototyping/` URL should lead to the non-go prototyping instructions — I have linked to it in various contexts, so it would not be great if we moved the content away.

oliverpool commented:
Back from holidays, I addressed your 2 comments.

I did a squash and a rebase on master (since you added the rsync-backup page in the meantime).

stapelberg commented:
Thanks very much! Now merged, should be live within a few minutes
